File: C++/insert-interval.cpp
class Solution
{
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval)
    {
        intervals.push_back(newInterval);
        return merge(intervals);
    }
    
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        
        vector<vector<int>> ans;
        
        ans.push_back(intervals[0]);
        
        for (int i = 1; i < intervals.size(); i++) {
            if (intervals[i][0] <= ans.back()[1] && intervals[i][1] >= ans.back()[0]) {
                ans.back()[0] = min(ans.back()[0], intervals[i][0]);
                ans.back()[1] = max(ans.back()[1], intervals[i][1]);
            } else {
                ans.push_back(intervals[i]);
            }
        }
        
        return ans;
    }
};

File: Go/insert-interval.go
func insert(intervals [][]int, newInterval []int) [][]int {
    // insert interval at expected position
    insertPosition, _ := slices.BinarySearchFunc(intervals, newInterval, func (a, b []int) int { 
        return a[0] - b[0] 
    })
    intervals = slices.Insert(intervals, insertPosition, newInterval)

    // merge any intervals that overlap
    result := [][]int{intervals[0]}
    last := 0
    for i := 1; i < len(intervals); i++ {
        if intervals[i][0] <= result[last][1] && intervals[i][1] >= result[last][0] {
            //result[last][0] = min(result[last][0], intervals[i][0])
            result[last][1] = max(result[last][1], intervals[i][1])
        } else {
            result = append(result, intervals[i])
            last++
        }
    }
    return result
}